src/Microsoft.DotNet.MSBuildSdkResolver/Interop.NETStandard.cs

```diff
@@ -16,6 +17,16 @@ internal static partial class Interop
     {
         internal static readonly bool RunningOnWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
 
+        internal static string realpath (string path)
+        {
+            var buffer = new StringBuilder(4096); // PATH_MAX from linux/limits.h
```

Comment:
I do feel strongly. We really must not leave buffer overflow to chance.

Also, the set of platforms that matters is the intersection of platforms supported by Mono and CLI. We won't find `dotnet` if there's no `dotnet` available for the platform.

Can we use `readlink(2)` instead? We'd have to loop if we need to support multiple levels of symlinks, but that API is at least safe by design as the buffer size is passed down.